### PR TITLE
Switched to a new dump functionality exposed by IQ#.

### PR DESCRIPTION
**Pull request by kuzminrobin**
Simplified the Dump functionality for QuantumSimulator and the functionality use in IQSharp.
This is a part of the set of 3 PRs:
[Q#RT side](https://github.com/microsoft/qsharp-runtime/pull/898),
[IQ#](https://github.com/microsoft/iqsharp/pull/562),
[Katas](https://github.com/microsoft/QuantumKatas/pull/727).

### CODE REVIEW

**Review by cgranade — APPROVED** [review]
This looks good to me, but would be good to get @tcNickolas' review as code owner as well. One thing I'll point out quickly, though, is that as written in this PR, the `%kata` magic command won't support histogram views for displayable states. I think that was already the case before this PR, so not a regression so much as a known limitation.

**Comment by tcNickolas** [comment]
@kuzminrobin I'm not very familiar with the katas magic part of code, @anjbur usually helps me with reviews on them. Can you please provide a bit of context on this change? Are there going to be any user-facing behavior changes? What about @cgranade's  comment on #688?

(Also, it looks like this PR supersedes #688 - could you please close the other PR to keep things more organized?)

**Comment by kuzminrobin** [comment]
> Can you please provide a bit of context on this change? 

This change stops calling the functionality (in IQ#) that is being removed as a simplification of the Dump functionality in QuantumSimulator. 
> Are there going to be any user-facing behavior changes?

No. Nothing that the user's code could be relying on.
> What about @cgranade's comment on #688?

I added a reply there that the note has been resolved in this repeated attempt.
> could you please close the other PR

Done. Thanks for the reminder! ;-)

**Review by tcNickolas — APPROVED** [review]
@cgranade Right, [this bug](https://github.com/microsoft/iqsharp/issues/450)! I thought initially that your comment was about the measurement probability bars shown in regular DumpMachine output, thank you for clarifying!

That was my one concern, I'll trust you folks on the syntax of the change :-) Approving, I'll let you update the branch to latest and merge when the time is right (not sure if it needs to go in sync with the other two PRs)

File: utilities/Microsoft.Quantum.Katas/KataMagic.cs [hunk 1]
```diff
@@ -206,8 +207,8 @@ public virtual bool Simulate(OperationInfo test, OperationInfo userAnswer, IChan
         /// Creates the instance of the simulator to use to run the Test 
         /// (for now always CounterSimulator from the same package).
         /// </summary>
-        public virtual SimulatorBase CreateSimulator(IChannel channel) =>
-            new CounterSimulator().WithJupyterDisplay(channel, ConfigurationSource);
+        public virtual SimulatorBase CreateSimulator() =>
+            new CounterSimulator();
```
Comment (DmitryVasilevsky) [inline utilities/Microsoft.Quantum.Katas/KataMagic.cs:211]: Cleans it up nicely!

**Review by DmitryVasilevsky — APPROVED** [review]

**Pull request merged**